packages/extension/src/contexts/LedgerContext/LedgerContext.tsx

```diff
@@ -178,69 +211,40 @@ const LedgerProvider: FC = ({ children }) => {
       const wallet = getCurrentWallet();
       if (!client) {
         throw new Error(LEDGER_CONNECTION_ERROR);
-      } else if (!wallet) {
-        throw new Error('You need to have a wallet connected to make a transaction');
-      } else {
-        // Prepare the transaction
-        const prepared = await client.autofill(transaction);
-        if (!prepared.Fee) {
-          throw new Error("Couldn't calculate the fees, something went wrong");
-        } else {
-          return prepared.Fee;
-        }
       }
-    },
-    [client, getCurrentWallet]
-  );
 
-  const getNFTs = useCallback(
-    async (payload?: GetNFTRequest): Promise<AccountNFTokenResponse> => {
-      const wallet = getCurrentWallet();
-      if (!client) {
-        throw new Error('You need to be connected to a ledger to get the NFTs');
-      }
-      if (!wallet) {
-        throw new Error('You need to have a wallet connected to get the NFTs');
+      if (!wallet?.seed) {
```

Comment:
Maybe you don't need to check the seed at all, how about:
```suggestion
      if (!wallet) {
```

Reply:
Right 👍